### DISCUSSION TITLE
declarative NixOS installation of the CoApp

### DISCUSSION THREAD

**Discussion by JJJHolscher** [aclap-dev/video-downloadhelper#1347] [Help and Q&A]
### Your question

Nix home-manager and NixOS focus on declarative intallations, so when I had to run a manual `vhdcoapp install` after installing the CoApp, I went looking for the Nix way to automate this and got some answers [on the nixpkgs github](https://github.com/NixOS/nixpkgs/issues/112046#issuecomment-2402266284). I haven't tried these myself yet, but I'll report back when I do.

I reckon this discussion place would not have the expertise to help me, but this might nonetheless be the place where fellow NixOS users might look for an answer, hence I posted this.

### Addon & CoApp version and details

```
version: 9.1.0.44
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"/nix/store/k5inwzpp6a0295pd3nfckk9hq8wmifhz-nodejs-20.15.1/bin/node","version":"2.0.19","new_version":false}
license: {"unneeded":true}
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:130.0) Gecko/20100101 Firefox/130.0
{
   [ configuration intentionally omitted ]
}
```
(man by default that clipboard option really over-shares, considering how github might already read whatever you paste before you submit, you could be unintentionally sharing your download history if you immediately paste into github before having looked at it)

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user . . . who knows diddly squat about Linux.

You are correct that the configuration information does contain a lot of irrelevant stuff.  You're on Linux so you don't have a license because there is no license on Linux.  But for those of us who do have a license, that information contains part of your license key & sometimes your E-mail address.  In later releases, the E-mail address is no longer shown, but the partial license is still shown.  I contend it is irrelevant to problem diagnosis.  Only the fact that you do or don't have a license is important.  There is NEVER a reason to share even a single character of your license key.

But I think you've removed more of the information than you really should have.  Here is my information, edited for security:

```
version: 9.1.0.41
target: mozilla
channel: beta
lang: en-US
coapp: {"found":true,"path":"C:\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.20","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"<removed>"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:131.0) Gecko/20100101 Firefox/131.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "M:\\VDH Testing",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "M:\\Adult Time"
        }
    },
    "concurrent_downloads_max": 32,
    "show_success_notification_for_icognito": true,
    "theme": "system",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": true,
        "hide_downloaded": false
    },
    "default_action": "download_as",
    "use_wide_ui": true,
    "use_legacy_ui": false,
    "use_sidebar": false,
    "open_count_store": 1,
    "successfull_dl": 69,
    "never_show_successfull_dl_message": true,
    "download_history": {
<removed>
    },
    "record_download_history": true,
    "history_limit_in_days": 2
}
```

It's really only the partial license & the download history that should be removed.  The rest of the information could be helpful in problem diagnosis.  Maybe.  You never know so you might include it.  I personally post my information like this:

`Windows 11 Pro 64-bit version 24H2 build 26100.2033, Firefox 131.0.2 64-bit, licensed VDH 9.1.0.41 beta, CoApp 2.0.20`

The information about Windows & Firefox in my abbreviated posting is more detailed than what is in the stuff generated from the VDH settings.  The other VDH settings are not often important.  Not often.  That's not to say never.  But not often.

I have been wondering about `"never_show_successfull_dl_message": true,`.  I don't know what that means & I don't see a setting for that.  Not to mention that the correct spelling of the word is `successful` with one letter l.  I always get the notification of a successful download.  It goes in the standard Windows notifications, the thing you can get by clicking the little bell in the bottom right corner of the Task Bar, the thing you can get with the `Win+n` key sequence.  So I don't know what that setting is.

**Reply by GrampaWildWilly** [reply]
And I commend your spirit of helpfulness.
